subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -255,6 +351,58 @@ static void oacp_read_proc_execute(struct bt_ots *ots,
 	}
 }
 
+#if defined(CONFIG_BT_OTS_OACP_WRITE_SUPPORT)
+static int oacp_write_proc_cb(struct bt_gatt_ots_l2cap *l2cap_ctx,
+			struct bt_conn *conn, struct net_buf *buf)
+{
+	struct bt_gatt_ots_object_write_op *write_op;
+	struct bt_ots *ots;
+	uint32_t offset;
+	uint32_t rem;
+	uint32_t len;
+	int rc;
+
+	ots = CONTAINER_OF(l2cap_ctx, struct bt_ots, l2cap);
+
+	if (!ots->cur_obj) {
+		LOG_ERR("Invalid Current Object on OACP Write procedure");
+		return -ENODEV;
+	}
+
+	if (!ots->cb->obj_write) {
+		ots->cur_obj->state.type = BT_GATT_OTS_OBJECT_IDLE_STATE;
+		LOG_ERR("OTS Write operation failed: "
+			"there is no OTS Write callback");
+		return -ENODEV;
+	}
+
+	write_op = &ots->cur_obj->state.write_op;
+	offset = write_op->oacp_params.offset + write_op->recv_len;
+	len = buf->len;
+	if (write_op->recv_len + len > write_op->oacp_params.len) {
+		LOG_WRN("More bytes received than the client indicated");
+		len = write_op->oacp_params.len - write_op->recv_len;
+	}
+	write_op->recv_len += len;
+	rem = write_op->oacp_params.len - write_op->recv_len;
+
+	if (write_op->recv_len == write_op->oacp_params.len) {
+		LOG_DBG("OACP Write Op over L2CAP is completed");
+
+		ots->cur_obj->state.type = BT_GATT_OTS_OBJECT_IDLE_STATE;
+		ots->l2cap.rx_done = NULL;
+	}
+
+	rc = ots->cb->obj_write(ots, conn, ots->cur_obj->id, buf->data, len,
+				  offset, rem);
+	if (rc == 0)
+		if (offset + len > ots->cur_obj->metadata.size.cur)
+			ots->cur_obj->metadata.size.cur = offset + len;
```

Comment:
Zephyr style guide states that even one-liner if statement shall use curly brackets

Reply:
I am surprised the style check passed :(! Sorry about that!